ecommerce-app/app/Http/Controllers/CartController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\ShoppingCartItem;
+use Str;
+use App\Models\ShoppingCart;
+use Illuminate\Http\Request;
+
+class CartController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(): ShoppingCart
+    {
+        $cart = new ShoppingCart();
+        $cart->id = Str::uuid();
+        $cart->user_id = auth()->user()->id;
+        $cart->total = 0;
+        $cart->save();
+
+        session()->put("card_id", $cart->id);
+
+        return $cart;
+    }
+
+    /**
+     * Display the specified resource.
+     */
+    public function show(string $id)
+    {
+        if (session("card_id")) {
+            $cart = ShoppingCart::findOrFail(session("card_id"));
+            $order_items = $cart->products()->orderBy("id", "desc")->get()->all();
+
+            return view("cart.show", ["cart" => $cart, "order_items" => $order_items]);
+        } else {
+            $cart = $this->store();
+            return view("cart.show", ["cart" => $cart, "order_items" => null]);
+        }
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     */
+    public function edit(string $id)
+    {
+        //
+    }
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request)
+    {
+        $card_id = null;
+        if (session("card_id")) {
+            $card_id = session("card_id");
+        } else {
+            $cart = $this->store();
+            $card_id = $cart->id;
+        }
```

Comment:
dùng toán tử 3 ngôi
$card_id = session("card_id") ?  session("card_id") : $this->store()->id;